File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n  = nums.size();
        sort(nums.begin(),nums.end());
        int c = 1;
        int p = nums[0];
        for(int i=1;i<n;i++)
        {
            if(nums[i] == nums[i-1])
            {
                c++;
                p = nums[i];
            }
            else
            {
                if(c > n/2)
                    break;
                c = 1;
                p = nums[i];
            }
        }
        return p;
    }
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int n = nums.length;
        int c = 0;
        int cand = 0;
        for(int i=0;i<n;i++)
        {
            if(c == 0)
            {
                cand = nums[i];
                c++;
            }
            else if(cand == nums[i])
            {
                
                c++;
                
            }
            else
            {
                c--;
            }
        }
        c=0;
        int p =0;
        for(int i=0;i<n;i++)
        {
            if(cand == nums[i])
            {
                p++;
            }
        }
        if(p>n/2)
            return cand;
        
        return -1;
        
    }
}